File: 0590/solution.go
package main

type Node struct {
	Val      int
	Children []*Node
}

func postorder(root *Node) []int {
	if root == nil {
		return []int{}
	}

	post := []int{}
	stack := []*Node{root}

	for len(stack) > 0 {
		curr := stack[len(stack)-1]
		stack = stack[:len(stack)-1]
		post = append(post, curr.Val)
		for _, node := range curr.Children {
			stack = append(stack, node)
		}
	}

	for i, j := 0, len(post)-1; i < j; i, j = i+1, j-1 {
		post[i], post[j] = post[j], post[i]
	}
	return post
}

File: 0590/solution.java
import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class Node {
    public int val;
    public List<Node> children;

    public Node() {
    }

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};

class Solution {
    public List<Integer> postorder(Node root) {
        List<Integer> post = new ArrayList<>();
        ArrayDeque<Node> stack = new ArrayDeque<>();
        if (root != null) {
            stack.add(root);
        }
        while (!stack.isEmpty()) {
            Node curr = stack.pollLast();
            post.add(curr.val);
            for (Node child : curr.children) {
                stack.add(child);
            }
        }
        Collections.reverse(post);
        return post;
    }
}

// time O(n)
// space O(1000)